src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,103 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderRequestBody } from '../../../order';
+import { PaymentMethodFailedError } from '../../errors';
+import Payment from '../../payment';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { GooglePayPaymentStrategy, PaymentMethodData } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+const TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+const LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+export default class CheckoutcomGooglePayPaymentStrategy extends GooglePayPaymentStrategy {
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._googlePayOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        }
+
+        const {payment: {methodId = ''} = {}} = payload;
+        const {paymentMethods: {getPaymentMethodOrThrow}} = this._store.getState();
+
+        const testMode = !!getPaymentMethodOrThrow(methodId).config.testMode;
+        const {checkoutcomkey} = getPaymentMethodOrThrow(methodId).initializationData;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this._googlePayPaymentProcessor.displayWallet()
+                        .then(paymentData => this._paymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+            })
+            .then(() => this._convertToken(testMode, checkoutcomkey))
```

Comment:
We should not use a mixture of `then/catch` and `async/await` in the same file. Please keep the usage consistent. I read your comment regarding `async/await` not defined in the parent class. First of all, I don't believe you should use inheritance. And secondly, even if you have no choice but to use inheritance, it shouldn't matter because `async/await` is simply static sugar for `then/catch`.

Reply:
as execute is an override i cant change it to async execute, so i changed every other async/await pair to plain promises, to keep it consistent as requested